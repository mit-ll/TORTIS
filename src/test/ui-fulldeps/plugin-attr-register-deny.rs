// aux-build:attr-plugin-test.rs
// ignore-stage1

#![feature(plugin)]
#![plugin(attr_plugin_test)]
//~^ WARN use of deprecated attribute `plugin`
#![deny(unused_attributes)]

#[baz]
fn baz() { } // no error

#[foo]
pub fn main() {
     //~^^ ERROR unused
    #[bar]
    fn inner() {}
    //~^^ ERROR crate
    //~^^^ ERROR unused
    baz();
    inner();
}
