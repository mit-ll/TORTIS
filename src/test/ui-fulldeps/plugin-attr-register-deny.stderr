warning: use of deprecated attribute `plugin`: compiler plugins are deprecated. See https://github.com/rust-lang/rust/pull/64675
  --> $DIR/plugin-attr-register-deny.rs:5:1
   |
LL | #![plugin(attr_plugin_test)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: may be removed in a future compiler version
   |
   = note: `#[warn(deprecated)]` on by default

error: unused attribute
  --> $DIR/plugin-attr-register-deny.rs:15:5
   |
LL |     #[bar]
   |     ^^^^^^
   |
note: lint level defined here
  --> $DIR/plugin-attr-register-deny.rs:7:9
   |
LL | #![deny(unused_attributes)]
   |         ^^^^^^^^^^^^^^^^^

error: crate-level attribute should be an inner attribute: add an exclamation mark: `#![foo]`
  --> $DIR/plugin-attr-register-deny.rs:15:5
   |
LL |     #[bar]
   |     ^^^^^^

error: unused attribute
  --> $DIR/plugin-attr-register-deny.rs:12:1
   |
LL | #[foo]
   | ^^^^^^

error: aborting due to 3 previous errors

